processing/src/main/java/org/apache/druid/segment/incremental/OffheapIncrementalIndex.java

```diff
@@ -346,4 +349,99 @@ public void close()
     }
     aggBuffers.clear();
   }
+
+  public static class Builder extends AppendableIndexBuilder
+  {
+    @Nullable
+    NonBlockingPool<ByteBuffer> bufferPool = null;
+
+    public Builder setBufferPool(final NonBlockingPool<ByteBuffer> bufferPool)
+    {
+      this.bufferPool = bufferPool;
+      return this;
+    }
+
+    @Override
+    public void validate()
+    {
+      super.validate();
+      if (bufferPool == null) {
+        throw new IllegalArgumentException("bufferPool cannot be null");
+      }
+    }
+
+    @Override
+    protected OffheapIncrementalIndex buildInner()
```

Comment:
this method should return IncrementalIndex
Is this legal to change the signature ?

Reply:
It is legal for an implementation to increase the requirement, but not to lower it.
As long as it conforms with the interface's contract.